# **Data Wrangling Project**

Breif Description

In [49]:
import numpy as np
import pandas as pd
import requests
import tweepy
import json
import matplotlib.pyplot as plt
import time

##  Data Gathering

In [50]:
#First Piece of Data(Twitter Archive)
DF_Twitter_Archive = pd.read_csv('twitter-archive-enhanced.csv')
DF_Twitter_Archive

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
0     892420643555336193                    NaN                  NaN   
1     892177421306343426                    NaN                  NaN   
2     891815181378084864                    NaN                  NaN   
3     891689557279858688                    NaN                  NaN   
4     891327558926688256                    NaN                  NaN   
...                  ...                    ...                  ...   
2351  666049248165822465                    NaN                  NaN   
2352  666044226329800704                    NaN                  NaN   
2353  666033412701032449                    NaN                  NaN   
2354  666029285002620928                    NaN                  NaN   
2355  666020888022790149                    NaN                  NaN   

                      timestamp  \
0     2017-08-01 16:23:56 +0000   
1     2017-08-01 00:17:27 +0000   
2     2017-07-31 00:18:03 +0000   
3     2017-07-30 15:58:51 +0000   
4     2017-07-29 16:00:24 +0000   
...                         ...   
2351  2015-11-16 00:24:50 +0000   
2352  2015-11-16 00:04:52 +0000   
2353  2015-11-15 23:21:54 +0000   
2354  2015-11-15 23:05:30 +0000   
2355  2015-11-15 22:32:08 +0000   

                                                 source  \
0     <a href="http://twitter.com/download/iphone" r...   
1     <a href="http://twitter.com/download/iphone" r...   
2     <a href="http://twitter.com/download/iphone" r...   
3     <a href="http://twitter.com/download/iphone" r...   
4     <a href="http://twitter.com/download/iphone" r...   
...                                                 ...   
2351  <a href="http://twitter.com/download/iphone" r...   
2352  <a href="http://twitter.com/download/iphone" r...   
2353  <a href="http://twitter.com/download/iphone" r...   
2354  <a href="http://twitter.com/download/iphone" r...   
2355  <a href="http://twitter.com/download/iphone" r...   

                                                   text  retweeted_status_id  \
0     This is Phineas. He's a mystical boy. Only eve...                  NaN   
1     This is Tilly. She's just checking pup on you....                  NaN   
2     This is Archie. He is a rare Norwegian Pouncin...                  NaN   
3     This is Darla. She commenced a snooze mid meal...                  NaN   
4     This is Franklin. He would like you to stop ca...                  NaN   
...                                                 ...                  ...   
2351  Here we have a 1949 1st generation vulpix. Enj...                  NaN   
2352  This is a purebred Piers Morgan. Loves to Netf...                  NaN   
2353  Here is a very happy pup. Big fan of well-main...                  NaN   
2354  This is a western brown Mitsubishi terrier. Up...                  NaN   
2355  Here we have a Japanese Irish Setter. Lost eye...                  NaN   

      retweeted_status_user_id retweeted_status_timestamp  \
0                          NaN                        NaN   
1                          NaN                        NaN   
2                          NaN                        NaN   
3                          NaN                        NaN   
4                          NaN                        NaN   
...                        ...                        ...   
2351                       NaN                        NaN   
2352                       NaN                        NaN   
2353                       NaN                        NaN   
2354                       NaN                        NaN   
2355                       NaN                        NaN   

                                          expanded_urls  rating_numerator  \
0     https://twitter.com/dog_rates/status/892420643...                13   
1     https://twitter.com/dog_rates/status/892177421...                13   
2     https://twitter.com/dog_rates/status/891815181...                12   
3     https://twitt

In [51]:
#Second Piece of Data(Image prediction )
URL = 'https://d17h27t6h515a5.cloudfront.net/topher/2017/August/599fd2ad_image-predictions/image-predictions.tsv'
with open('image_prediction_file.tsv','wb')as file:
    image_prediction = requests.get(URL)
    file.write(image_prediction.content)

df_image_pred = pd.read_csv('image_prediction_file.tsv',sep='\t')

df_image_pred

tweet_id                                          jpg_url  \
0     666020888022790149  https://pbs.twimg.com/media/CT4udn0WwAA0aMy.jpg   
1     666029285002620928  https://pbs.twimg.com/media/CT42GRgUYAA5iDo.jpg   
2     666033412701032449  https://pbs.twimg.com/media/CT4521TWwAEvMyu.jpg   
3     666044226329800704  https://pbs.twimg.com/media/CT5Dr8HUEAA-lEu.jpg   
4     666049248165822465  https://pbs.twimg.com/media/CT5IQmsXIAAKY4A.jpg   
...                  ...                                              ...   
2070  891327558926688256  https://pbs.twimg.com/media/DF6hr6BUMAAzZgT.jpg   
2071  891689557279858688  https://pbs.twimg.com/media/DF_q7IAWsAEuuN8.jpg   
2072  891815181378084864  https://pbs.twimg.com/media/DGBdLU1WsAANxJ9.jpg   
2073  892177421306343426  https://pbs.twimg.com/media/DGGmoV4XsAAUL6n.jpg   
2074  892420643555336193  https://pbs.twimg.com/media/DGKD1-bXoAAIAUK.jpg   

      img_num                      p1   p1_conf  p1_dog                  p2  \
0           1  Welsh_springer_spaniel  0.465074    True              collie   
1           1                 redbone  0.506826    True  miniature_pinscher   
2           1         German_shepherd  0.596461    True            malinois   
3           1     Rhodesian_ridgeback  0.408143    True             redbone   
4           1      miniature_pinscher  0.560311    True          Rottweiler   
...       ...                     ...       ...     ...                 ...   
2070        2                  basset  0.555712    True    English_springer   
2071        1             paper_towel  0.170278   False  Labrador_retriever   
2072        1               Chihuahua  0.716012    True            malamute   
2073        1               Chihuahua  0.323581    True            Pekinese   
2074        1                  orange  0.097049   False               bagel   

       p2_conf  p2_dog                           p3   p3_conf  p3_dog  
0     0.156665    True            Shetland_sheepdog  0.061428    True  
1     0.074192    True          Rhodesian_ridgeback  0.072010    True  
2     0.138584    True                   bloodhound  0.116197    True  
3     0.360687    True           miniature_pinscher  0.222752    True  
4     0.243682    True                     Doberman  0.154629    True  
...        ...     ...                          ...       ...     ...  
2070  0.225770    True  German_short-haired_pointer  0.175219    True  
2071  0.168086    True                      spatula  0.040836   False  
2072  0.078253    True                       kelpie  0.031379    True  
2073  0.090647    True                     papillon  0.068957    True  
2074  0.085851   False                       banana  0.076110   False  

[2075 rows x 12 columns]

In [52]:
#Third Piece of Data(API collected tweets Data)
DF_tweets_data = pd.read_json('tweet-json.txt',lines=True, encoding = 'utf-8')

In [53]:
DF_tweets_data.head(5)


created_at                  id              id_str  \
0 2017-08-01 16:23:56+00:00  892420643555336193  892420643555336192   
1 2017-08-01 00:17:27+00:00  892177421306343426  892177421306343424   
2 2017-07-31 00:18:03+00:00  891815181378084864  891815181378084864   
3 2017-07-30 15:58:51+00:00  891689557279858688  891689557279858688   
4 2017-07-29 16:00:24+00:00  891327558926688256  891327558926688256   

                                           full_text  truncated  \
0  This is Phineas. He's a mystical boy. Only eve...      False   
1  This is Tilly. She's just checking pup on you....      False   
2  This is Archie. He is a rare Norwegian Pouncin...      False   
3  This is Darla. She commenced a snooze mid meal...      False   
4  This is Franklin. He would like you to stop ca...      False   

  display_text_range                                           entities  \
0            [0, 85]  {'hashtags': [], 'symbols': [], 'user_mentions...   
1           [0, 138]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2           [0, 121]  {'hashtags': [], 'symbols': [], 'user_mentions...   
3            [0, 79]  {'hashtags': [], 'symbols': [], 'user_mentions...   
4           [0, 138]  {'hashtags': [{'text': 'BarkWeek', 'indices': ...   

                                   extended_entities  \
0  {'media': [{'id': 892420639486877696, 'id_str'...   
1  {'media': [{'id': 892177413194625024, 'id_str'...   
2  {'media': [{'id': 891815175371796480, 'id_str'...   
3  {'media': [{'id': 891689552724799489, 'id_str'...   
4  {'media': [{'id': 891327551943041024, 'id_str'...   

                                              source  in_reply_to_status_id  \
0  <a href="http://twitter.com/download/iphone" r...                    NaN   
1  <a href="http://twitter.com/download/iphone" r...                    NaN   
2  <a href="http://twitter.com/download/iphone" r...                    NaN   
3  <a href="http://twitter.com/download/iphone" r...                    NaN   
4  <a href="http://twitter.com/download/iphone" r...                    NaN   

   ...  favorite_count  favorited  retweeted possibly_sensitive  \
0  ...           39467      False      False                0.0   
1  ...           33819      False      False                0.0   
2  ...           25461      False      False                0.0   
3  ...           42908      False      False                0.0   
4  ...           41048      False      False                0.0   

  possibly_sensitive_appealable  lang  retweeted_status quoted_status_id  \
0                           0.0    en               NaN              NaN   
1                           0.0    en               NaN              NaN   
2                           0.0    en               NaN              NaN   
3                           0.0    en               NaN              NaN   
4                           0.0    en               NaN              NaN   

   quoted_status_id_str  quoted_status  
0                   NaN            NaN  
1                   NaN            NaN  
2                   NaN            NaN  
3                   NaN            NaN  
4                   NaN            NaN  

[5 rows x 31 columns]

##  Data Assessment

### Visual Assessment 

In [54]:
DF_Twitter_Archive.head()

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
0  892420643555336193                    NaN                  NaN   
1  892177421306343426                    NaN                  NaN   
2  891815181378084864                    NaN                  NaN   
3  891689557279858688                    NaN                  NaN   
4  891327558926688256                    NaN                  NaN   

                   timestamp  \
0  2017-08-01 16:23:56 +0000   
1  2017-08-01 00:17:27 +0000   
2  2017-07-31 00:18:03 +0000   
3  2017-07-30 15:58:51 +0000   
4  2017-07-29 16:00:24 +0000   

                                              source  \
0  <a href="http://twitter.com/download/iphone" r...   
1  <a href="http://twitter.com/download/iphone" r...   
2  <a href="http://twitter.com/download/iphone" r...   
3  <a href="http://twitter.com/download/iphone" r...   
4  <a href="http://twitter.com/download/iphone" r...   

                                                text  retweeted_status_id  \
0  This is Phineas. He's a mystical boy. Only eve...                  NaN   
1  This is Tilly. She's just checking pup on you....                  NaN   
2  This is Archie. He is a rare Norwegian Pouncin...                  NaN   
3  This is Darla. She commenced a snooze mid meal...                  NaN   
4  This is Franklin. He would like you to stop ca...                  NaN   

   retweeted_status_user_id retweeted_status_timestamp  \
0                       NaN                        NaN   
1                       NaN                        NaN   
2                       NaN                        NaN   
3                       NaN                        NaN   
4                       NaN                        NaN   

                                       expanded_urls  rating_numerator  \
0  https://twitter.com/dog_rates/status/892420643...                13   
1  https://twitter.com/dog_rates/status/892177421...                13   
2  https://twitter.com/dog_rates/status/891815181...                12   
3  https://twitter.com/dog_rates/status/891689557...                13   
4  https://twitter.com/dog_rates/status/891327558...                12   

   rating_denominator      name doggo floofer pupper puppo  
0                  10   Phineas  None    None   None  None  
1                  10     Tilly  None    None   None  None  
2                  10    Archie  None    None   None  None  
3                  10     Darla  None    None   None  None  
4                  10  Franklin  None    None   None  None

In [55]:
DF_Twitter_Archive.tail()

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
2351  666049248165822465                    NaN                  NaN   
2352  666044226329800704                    NaN                  NaN   
2353  666033412701032449                    NaN                  NaN   
2354  666029285002620928                    NaN                  NaN   
2355  666020888022790149                    NaN                  NaN   

                      timestamp  \
2351  2015-11-16 00:24:50 +0000   
2352  2015-11-16 00:04:52 +0000   
2353  2015-11-15 23:21:54 +0000   
2354  2015-11-15 23:05:30 +0000   
2355  2015-11-15 22:32:08 +0000   

                                                 source  \
2351  <a href="http://twitter.com/download/iphone" r...   
2352  <a href="http://twitter.com/download/iphone" r...   
2353  <a href="http://twitter.com/download/iphone" r...   
2354  <a href="http://twitter.com/download/iphone" r...   
2355  <a href="http://twitter.com/download/iphone" r...   

                                                   text  retweeted_status_id  \
2351  Here we have a 1949 1st generation vulpix. Enj...                  NaN   
2352  This is a purebred Piers Morgan. Loves to Netf...                  NaN   
2353  Here is a very happy pup. Big fan of well-main...                  NaN   
2354  This is a western brown Mitsubishi terrier. Up...                  NaN   
2355  Here we have a Japanese Irish Setter. Lost eye...                  NaN   

      retweeted_status_user_id retweeted_status_timestamp  \
2351                       NaN                        NaN   
2352                       NaN                        NaN   
2353                       NaN                        NaN   
2354                       NaN                        NaN   
2355                       NaN                        NaN   

                                          expanded_urls  rating_numerator  \
2351  https://twitter.com/dog_rates/status/666049248...                 5   
2352  https://twitter.com/dog_rates/status/666044226...                 6   
2353  https://twitter.com/dog_rates/status/666033412...                 9   
2354  https://twitter.com/dog_rates/status/666029285...                 7   
2355  https://twitter.com/dog_rates/status/666020888...                 8   

      rating_denominator  name doggo floofer pupper puppo  
2351                  10  None  None    None   None  None  
2352                  10     a  None    None   None  None  
2353                  10     a  None    None   None  None  
2354                  10     a  None    None   None  None  
2355                  10  None  None    None   None  None

In [56]:
DF_Twitter_Archive.sample()

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
1501  692142790915014657           6.920419e+17         4.196984e+09   

                      timestamp  \
1501  2016-01-27 00:31:15 +0000   

                                                 source  \
1501  <a href="http://twitter.com/download/iphone" r...   

                                                   text  retweeted_status_id  \
1501  These are some pictures of Teddy that further ...                  NaN   

      retweeted_status_user_id retweeted_status_timestamp  \
1501                       NaN                        NaN   

                                          expanded_urls  rating_numerator  \
1501  https://twitter.com/dog_rates/status/692142790...                13   

      rating_denominator  name doggo floofer pupper puppo  
1501                  10  None  None    None   None  None

In [57]:
df_image_pred.head()

tweet_id                                          jpg_url  \
0  666020888022790149  https://pbs.twimg.com/media/CT4udn0WwAA0aMy.jpg   
1  666029285002620928  https://pbs.twimg.com/media/CT42GRgUYAA5iDo.jpg   
2  666033412701032449  https://pbs.twimg.com/media/CT4521TWwAEvMyu.jpg   
3  666044226329800704  https://pbs.twimg.com/media/CT5Dr8HUEAA-lEu.jpg   
4  666049248165822465  https://pbs.twimg.com/media/CT5IQmsXIAAKY4A.jpg   

   img_num                      p1   p1_conf  p1_dog                  p2  \
0        1  Welsh_springer_spaniel  0.465074    True              collie   
1        1                 redbone  0.506826    True  miniature_pinscher   
2        1         German_shepherd  0.596461    True            malinois   
3        1     Rhodesian_ridgeback  0.408143    True             redbone   
4        1      miniature_pinscher  0.560311    True          Rottweiler   

    p2_conf  p2_dog                   p3   p3_conf  p3_dog  
0  0.156665    True    Shetland_sheepdog  0.061428    True  
1  0.074192    True  Rhodesian_ridgeback  0.072010    True  
2  0.138584    True           bloodhound  0.116197    True  
3  0.360687    True   miniature_pinscher  0.222752    True  
4  0.243682    True             Doberman  0.154629    True

In [58]:
df_image_pred.tail()

tweet_id                                          jpg_url  \
2070  891327558926688256  https://pbs.twimg.com/media/DF6hr6BUMAAzZgT.jpg   
2071  891689557279858688  https://pbs.twimg.com/media/DF_q7IAWsAEuuN8.jpg   
2072  891815181378084864  https://pbs.twimg.com/media/DGBdLU1WsAANxJ9.jpg   
2073  892177421306343426  https://pbs.twimg.com/media/DGGmoV4XsAAUL6n.jpg   
2074  892420643555336193  https://pbs.twimg.com/media/DGKD1-bXoAAIAUK.jpg   

      img_num           p1   p1_conf  p1_dog                  p2   p2_conf  \
2070        2       basset  0.555712    True    English_springer  0.225770   
2071        1  paper_towel  0.170278   False  Labrador_retriever  0.168086   
2072        1    Chihuahua  0.716012    True            malamute  0.078253   
2073        1    Chihuahua  0.323581    True            Pekinese  0.090647   
2074        1       orange  0.097049   False               bagel  0.085851   

      p2_dog                           p3   p3_conf  p3_dog  
2070    True  German_short-haired_pointer  0.175219    True  
2071    True                      spatula  0.040836   False  
2072    True                       kelpie  0.031379    True  
2073    True                     papillon  0.068957    True  
2074   False                       banana  0.076110   False

In [59]:
df_image_pred.sample()

tweet_id                                          jpg_url  \
1202  741303864243200000  https://pbs.twimg.com/media/Ckmj7mNWYAA4NzZ.jpg   

      img_num         p1   p1_conf  p1_dog   p2   p2_conf  p2_dog        p3  \
1202        1  Chihuahua  0.768156    True  pug  0.014902    True  Pekinese   

       p3_conf  p3_dog  
1202  0.012816    True

In [60]:
DF_tweets_data.head()

created_at                  id              id_str  \
0 2017-08-01 16:23:56+00:00  892420643555336193  892420643555336192   
1 2017-08-01 00:17:27+00:00  892177421306343426  892177421306343424   
2 2017-07-31 00:18:03+00:00  891815181378084864  891815181378084864   
3 2017-07-30 15:58:51+00:00  891689557279858688  891689557279858688   
4 2017-07-29 16:00:24+00:00  891327558926688256  891327558926688256   

                                           full_text  truncated  \
0  This is Phineas. He's a mystical boy. Only eve...      False   
1  This is Tilly. She's just checking pup on you....      False   
2  This is Archie. He is a rare Norwegian Pouncin...      False   
3  This is Darla. She commenced a snooze mid meal...      False   
4  This is Franklin. He would like you to stop ca...      False   

  display_text_range                                           entities  \
0            [0, 85]  {'hashtags': [], 'symbols': [], 'user_mentions...   
1           [0, 138]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2           [0, 121]  {'hashtags': [], 'symbols': [], 'user_mentions...   
3            [0, 79]  {'hashtags': [], 'symbols': [], 'user_mentions...   
4           [0, 138]  {'hashtags': [{'text': 'BarkWeek', 'indices': ...   

                                   extended_entities  \
0  {'media': [{'id': 892420639486877696, 'id_str'...   
1  {'media': [{'id': 892177413194625024, 'id_str'...   
2  {'media': [{'id': 891815175371796480, 'id_str'...   
3  {'media': [{'id': 891689552724799489, 'id_str'...   
4  {'media': [{'id': 891327551943041024, 'id_str'...   

                                              source  in_reply_to_status_id  \
0  <a href="http://twitter.com/download/iphone" r...                    NaN   
1  <a href="http://twitter.com/download/iphone" r...                    NaN   
2  <a href="http://twitter.com/download/iphone" r...                    NaN   
3  <a href="http://twitter.com/download/iphone" r...                    NaN   
4  <a href="http://twitter.com/download/iphone" r...                    NaN   

   ...  favorite_count  favorited  retweeted possibly_sensitive  \
0  ...           39467      False      False                0.0   
1  ...           33819      False      False                0.0   
2  ...           25461      False      False                0.0   
3  ...           42908      False      False                0.0   
4  ...           41048      False      False                0.0   

  possibly_sensitive_appealable  lang  retweeted_status quoted_status_id  \
0                           0.0    en               NaN              NaN   
1                           0.0    en               NaN              NaN   
2                           0.0    en               NaN              NaN   
3                           0.0    en               NaN              NaN   
4                           0.0    en               NaN              NaN   

   quoted_status_id_str  quoted_status  
0                   NaN            NaN  
1                   NaN            NaN  
2                   NaN            NaN  
3                   NaN            NaN  
4                   NaN            NaN  

[5 rows x 31 columns]

In [61]:
DF_tweets_data.tail()

created_at                  id              id_str  \
2349 2015-11-16 00:24:50+00:00  666049248165822465  666049248165822464   
2350 2015-11-16 00:04:52+00:00  666044226329800704  666044226329800704   
2351 2015-11-15 23:21:54+00:00  666033412701032449  666033412701032448   
2352 2015-11-15 23:05:30+00:00  666029285002620928  666029285002620928   
2353 2015-11-15 22:32:08+00:00  666020888022790149  666020888022790144   

                                              full_text  truncated  \
2349  Here we have a 1949 1st generation vulpix. Enj...      False   
2350  This is a purebred Piers Morgan. Loves to Netf...      False   
2351  Here is a very happy pup. Big fan of well-main...      False   
2352  This is a western brown Mitsubishi terrier. Up...      False   
2353  Here we have a Japanese Irish Setter. Lost eye...      False   

     display_text_range                                           entities  \
2349           [0, 120]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2350           [0, 137]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2351           [0, 130]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2352           [0, 139]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2353           [0, 131]  {'hashtags': [], 'symbols': [], 'user_mentions...   

                                      extended_entities  \
2349  {'media': [{'id': 666049244999131136, 'id_str'...   
2350  {'media': [{'id': 666044217047650304, 'id_str'...   
2351  {'media': [{'id': 666033409081393153, 'id_str'...   
2352  {'media': [{'id': 666029276303482880, 'id_str'...   
2353  {'media': [{'id': 666020881337073664, 'id_str'...   

                                                 source  \
2349  <a href="http://twitter.com/download/iphone" r...   
2350  <a href="http://twitter.com/download/iphone" r...   
2351  <a href="http://twitter.com/download/iphone" r...   
2352  <a href="http://twitter.com/download/iphone" r...   
2353  <a href="http://twitter.com/download/iphone" r...   

      in_reply_to_status_id  ...  favorite_count  favorited  retweeted  \
2349                    NaN  ...             111      False      False   
2350                    NaN  ...             311      False      False   
2351                    NaN  ...             128      False      False   
2352                    NaN  ...             132      False      False   
2353                    NaN  ...            2535      False      False   

     possibly_sensitive possibly_sensitive_appealable  lang  retweeted_status  \
2349                0.0                           0.0    en               NaN   
2350                0.0                           0.0    en               NaN   
2351                0.0                           0.0    en               NaN   
2352                0.0                           0.0    en               NaN   
2353                0.0                           0.0    en               NaN   

     quoted_status_id  quoted_status_id_str  quoted_status  
2349              NaN                   NaN            NaN  
2350              NaN                   NaN            NaN  
2351              NaN                   NaN            NaN  
2352              NaN                   NaN            NaN  
2353              NaN                   NaN            NaN  

[5 rows x 31 columns]

In [62]:
DF_tweets_data.sample()

created_at                  id              id_str  \
985 2016-07-02 00:27:45+00:00  749036806121881602  749036806121881600   

                                             full_text  truncated  \
985  This is Dietrich. He hops at random. Other dog...      False   

    display_text_range                                           entities  \
985           [0, 113]  {'hashtags': [], 'symbols': [], 'user_mentions...   

                                     extended_entities  \
985  {'media': [{'id': 749036286288207872, 'id_str'...   

                                                source  in_reply_to_status_id  \
985  <a href="http://twitter.com/download/iphone" r...                    NaN   

     ...  favorite_count  favorited  retweeted possibly_sensitive  \
985  ...            3425      False      False                0.0   

    possibly_sensitive_appealable  lang  retweeted_status quoted_status_id  \
985                           0.0    en               NaN              NaN   

     quoted_status_id_str  quoted_status  
985                   NaN            NaN  

[1 rows x 31 columns]

### Programmatic Assessment 

In [63]:
DF_Twitter_Archive.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2356 entries, 0 to 2355
Data columns (total 17 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   tweet_id                    2356 non-null   int64  
 1   in_reply_to_status_id       78 non-null     float64
 2   in_reply_to_user_id         78 non-null     float64
 3   timestamp                   2356 non-null   object 
 4   source                      2356 non-null   object 
 5   text                        2356 non-null   object 
 6   retweeted_status_id         181 non-null    float64
 7   retweeted_status_user_id    181 non-null    float64
 8   retweeted_status_timestamp  181 non-null    object 
 9   expanded_urls               2297 non-null   object 
 10  rating_numerator            2356 non-null   int64  
 11  rating_denominator          2356 non-null   int64  
 12  name                        2356 non-null   object 
 13  doggo                       2356 

In [64]:
DF_Twitter_Archive.describe()

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
count  2.356000e+03           7.800000e+01         7.800000e+01   
mean   7.427716e+17           7.455079e+17         2.014171e+16   
std    6.856705e+16           7.582492e+16         1.252797e+17   
min    6.660209e+17           6.658147e+17         1.185634e+07   
25%    6.783989e+17           6.757419e+17         3.086374e+08   
50%    7.196279e+17           7.038708e+17         4.196984e+09   
75%    7.993373e+17           8.257804e+17         4.196984e+09   
max    8.924206e+17           8.862664e+17         8.405479e+17   

       retweeted_status_id  retweeted_status_user_id  rating_numerator  \
count         1.810000e+02              1.810000e+02       2356.000000   
mean          7.720400e+17              1.241698e+16         13.126486   
std           6.236928e+16              9.599254e+16         45.876648   
min           6.661041e+17              7.832140e+05          0.000000   
25%           7.186315e+17              4.196984e+09         10.000000   
50%           7.804657e+17              4.196984e+09         11.000000   
75%           8.203146e+17              4.196984e+09         12.000000   
max           8.874740e+17              7.874618e+17       1776.000000   

       rating_denominator  
count         2356.000000  
mean            10.455433  
std              6.745237  
min              0.000000  
25%             10.000000  
50%             10.000000  
75%             10.000000  
max            170.000000

In [65]:
DF_tweets_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2354 entries, 0 to 2353
Data columns (total 31 columns):
 #   Column                         Non-Null Count  Dtype              
---  ------                         --------------  -----              
 0   created_at                     2354 non-null   datetime64[ns, UTC]
 1   id                             2354 non-null   int64              
 2   id_str                         2354 non-null   int64              
 3   full_text                      2354 non-null   object             
 4   truncated                      2354 non-null   bool               
 5   display_text_range             2354 non-null   object             
 6   entities                       2354 non-null   object             
 7   extended_entities              2073 non-null   object             
 8   source                         2354 non-null   object             
 9   in_reply_to_status_id          78 non-null     float64            
 10  in_reply_to_status_id_st

In [66]:
DF_tweets_data.describe()

id        id_str  in_reply_to_status_id  \
count  2.354000e+03  2.354000e+03           7.800000e+01   
mean   7.426978e+17  7.426978e+17           7.455079e+17   
std    6.852812e+16  6.852812e+16           7.582492e+16   
min    6.660209e+17  6.660209e+17           6.658147e+17   
25%    6.783975e+17  6.783975e+17           6.757419e+17   
50%    7.194596e+17  7.194596e+17           7.038708e+17   
75%    7.993058e+17  7.993058e+17           8.257804e+17   
max    8.924206e+17  8.924206e+17           8.862664e+17   

       in_reply_to_status_id_str  in_reply_to_user_id  \
count               7.800000e+01         7.800000e+01   
mean                7.455079e+17         2.014171e+16   
std                 7.582492e+16         1.252797e+17   
min                 6.658147e+17         1.185634e+07   
25%                 6.757419e+17         3.086374e+08   
50%                 7.038708e+17         4.196984e+09   
75%                 8.257804e+17         4.196984e+09   
max                 8.862664e+17         8.405479e+17   

       in_reply_to_user_id_str  geo  coordinates  contributors  retweet_count  \
count             7.800000e+01  0.0          0.0           0.0    2354.000000   
mean              2.014171e+16  NaN          NaN           NaN    3164.797366   
std               1.252797e+17  NaN          NaN           NaN    5284.770364   
min               1.185634e+07  NaN          NaN           NaN       0.000000   
25%               3.086374e+08  NaN          NaN           NaN     624.500000   
50%               4.196984e+09  NaN          NaN           NaN    1473.500000   
75%               4.196984e+09  NaN          NaN           NaN    3652.000000   
max               8.405479e+17  NaN          NaN           NaN   79515.000000   

       favorite_count  possibly_sensitive  possibly_sensitive_appealable  \
count     2354.000000              2211.0                         2211.0   
mean      8080.968564                 0.0                            0.0   
std      11814.771334                 0.0                            0.0   
min          0.000000                 0.0                            0.0   
25%       1415.000000                 0.0                            0.0   
50%       3603.500000                 0.0                            0.0   
75%      10122.250000                 0.0                            0.0   
max     132810.000000                 0.0                            0.0   

       quoted_status_id  quoted_status_id_str  
count      2.900000e+01          2.900000e+01  
mean       8.162686e+17          8.162686e+17  
std        6.164161e+16          6.164161e+16  
min        6.721083e+17          6.721083e+17  
25%        7.888183e+17          7.888183e+17  
50%        8.340867e+17          8.340867e+17  
75%        8.664587e+17          8.664587e+17  
max        8.860534e+17          8.860534e+17

In [67]:
df_image_pred.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2075 entries, 0 to 2074
Data columns (total 12 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   tweet_id  2075 non-null   int64  
 1   jpg_url   2075 non-null   object 
 2   img_num   2075 non-null   int64  
 3   p1        2075 non-null   object 
 4   p1_conf   2075 non-null   float64
 5   p1_dog    2075 non-null   bool   
 6   p2        2075 non-null   object 
 7   p2_conf   2075 non-null   float64
 8   p2_dog    2075 non-null   bool   
 9   p3        2075 non-null   object 
 10  p3_conf   2075 non-null   float64
 11  p3_dog    2075 non-null   bool   
dtypes: bool(3), float64(3), int64(2), object(4)
memory usage: 152.1+ KB


In [68]:
df_image_pred.describe()

tweet_id      img_num      p1_conf       p2_conf       p3_conf
count  2.075000e+03  2075.000000  2075.000000  2.075000e+03  2.075000e+03
mean   7.384514e+17     1.203855     0.594548  1.345886e-01  6.032417e-02
std    6.785203e+16     0.561875     0.271174  1.006657e-01  5.090593e-02
min    6.660209e+17     1.000000     0.044333  1.011300e-08  1.740170e-10
25%    6.764835e+17     1.000000     0.364412  5.388625e-02  1.622240e-02
50%    7.119988e+17     1.000000     0.588230  1.181810e-01  4.944380e-02
75%    7.932034e+17     1.000000     0.843855  1.955655e-01  9.180755e-02
max    8.924206e+17     4.000000     1.000000  4.880140e-01  2.734190e-01

In [69]:
DF_TA_clean = DF_Twitter_Archive.copy() # DataFrame_Twitter Archive_clean
DF_IP_clean = df_image_pred.copy()      # DataFrame_Image Prediction_clean
DF_TD_clean = DF_tweets_data.copy()     # DataFrame_Twitter Data_clean

In [70]:
DF_TA_clean.duplicated().sum()

0

In [71]:
DF_IP_clean.duplicated().sum()

0

In [72]:
#DF_TA_clean.value_counts()

In [73]:
#DF_IP_clean.value_counts()

In [74]:
#DF_TD_clean.value_counts()

### Assessment Summary

#### Quality Issues

##### Twitter Archive table 
* 1- timestamp in string not in datetime format
* 2- +0000 in timestamp column entries 
* 5- letters in name column
* 6- unnecessary rows with retweeted data
* 7- unnecessary rows with reply data


##### Image Prediction table
* 8- p1, p2 and p3 names are misleading 
- (3-4) 2356 rows in Twitter archive, 2075 in image prediction and 2354 in tweets data(missing data)

#### Tweets Data
* 9- id column name should be tweet_id




### Tidiness Issues

##### Archive table 
* 1- Unnecessary columns  i.e  in_reply_to_status_id, in_reply_to_user_id empty and retweeted_status_id retweeted_status_user_id and retweeted_status_timestamp.

##### Tweet Data Table
* 2- Unnecessary columns like 'created at' which is similar to timestamp.


##  Data Cleaning

In [75]:
# 1-2 converting timestamp type from string to datetime and remove the hour Zeros
DF_TA_clean['timestamp'] = pd.to_datetime(DF_TA_clean.timestamp).dt.date

In [76]:
#Test
DF_TA_clean.head(1)

tweet_id  in_reply_to_status_id  in_reply_to_user_id   timestamp  \
0  892420643555336193                    NaN                  NaN  2017-08-01   

                                              source  \
0  <a href="http://twitter.com/download/iphone" r...   

                                                text  retweeted_status_id  \
0  This is Phineas. He's a mystical boy. Only eve...                  NaN   

   retweeted_status_user_id retweeted_status_timestamp  \
0                       NaN                        NaN   

                                       expanded_urls  rating_numerator  \
0  https://twitter.com/dog_rates/status/892420643...                13   

   rating_denominator     name doggo floofer pupper puppo  
0                  10  Phineas  None    None   None  None

In [77]:
# 3-4 removing excess data from Twitter Archive and Tweet_data
DF_TA_clean = DF_TA_clean.loc[:len(DF_IP_clean['tweet_id']),:]
DF_TD_clean = DF_TD_clean.loc[:len(DF_IP_clean['tweet_id']),:]

In [78]:
#Test
DF_TA_clean

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
0     892420643555336193                    NaN                  NaN   
1     892177421306343426                    NaN                  NaN   
2     891815181378084864                    NaN                  NaN   
3     891689557279858688                    NaN                  NaN   
4     891327558926688256                    NaN                  NaN   
...                  ...                    ...                  ...   
2071  671115716440031232                    NaN                  NaN   
2072  671109016219725825                    NaN                  NaN   
2073  670995969505435648                    NaN                  NaN   
2074  670842764863651840                    NaN                  NaN   
2075  670840546554966016                    NaN                  NaN   

       timestamp                                             source  \
0     2017-08-01  <a href="http://twitter.com/download/iphone" r...   
1     2017-08-01  <a href="http://twitter.com/download/iphone" r...   
2     2017-07-31  <a href="http://twitter.com/download/iphone" r...   
3     2017-07-30  <a href="http://twitter.com/download/iphone" r...   
4     2017-07-29  <a href="http://twitter.com/download/iphone" r...   
...          ...                                                ...   
2071  2015-11-29  <a href="http://twitter.com/download/iphone" r...   
2072  2015-11-29  <a href="http://twitter.com/download/iphone" r...   
2073  2015-11-29  <a href="http://twitter.com/download/iphone" r...   
2074  2015-11-29  <a href="http://twitter.com/download/iphone" r...   
2075  2015-11-29  <a href="http://twitter.com/download/iphone" r...   

                                                   text  retweeted_status_id  \
0     This is Phineas. He's a mystical boy. Only eve...                  NaN   
1     This is Tilly. She's just checking pup on you....                  NaN   
2     This is Archie. He is a rare Norwegian Pouncin...                  NaN   
3     This is Darla. She commenced a snooze mid meal...                  NaN   
4     This is Franklin. He would like you to stop ca...                  NaN   
...                                                 ...                  ...   
2071  Meet Phred. He isn't steering, looking at the ...                  NaN   
2072  This is Toby. He asked for chocolate cake for ...                  NaN   
2073  Yea I can't handle this job anymore your dogs ...                  NaN   
2074  After so many requests... here you go.\n\nGood...                  NaN   
2075  Meet Colby. He's that one cool friend that get...                  NaN   

      retweeted_status_user_id retweeted_status_timestamp  \
0                          NaN                        NaN   
1                          NaN                        NaN   
2                          NaN                        NaN   
3                          NaN                        NaN   
4                          NaN                        NaN   
...                        ...                        ...   
2071                       NaN                        NaN   
2072                       NaN                        NaN   
2073                       NaN                        NaN   
2074                       NaN                        NaN   
2075                       NaN                        NaN   

                                          expanded_urls  rating_numerator  \
0     https://twitter.com/dog_rates/status/892420643...                13   
1     https://twitter.com/dog_rates/status/892177421...                13   
2     https://twitter.com/dog_rates/status/891815181...                12   
3     https://twitter.com/dog_rates/status/891689557...                13   
4     https://twitter.com/dog_rates/status/891327558...                12   
...                                                 ...               ...   
2071  https://twitter.com/dog_rates/status/671115716...          

In [79]:
#Test
DF_TD_clean

created_at                  id              id_str  \
0    2017-08-01 16:23:56+00:00  892420643555336193  892420643555336192   
1    2017-08-01 00:17:27+00:00  892177421306343426  892177421306343424   
2    2017-07-31 00:18:03+00:00  891815181378084864  891815181378084864   
3    2017-07-30 15:58:51+00:00  891689557279858688  891689557279858688   
4    2017-07-29 16:00:24+00:00  891327558926688256  891327558926688256   
...                        ...                 ...                 ...   
2071 2015-11-29 16:01:20+00:00  670995969505435648  670995969505435648   
2072 2015-11-29 05:52:33+00:00  670842764863651840  670842764863651840   
2073 2015-11-29 05:43:44+00:00  670840546554966016  670840546554966016   
2074 2015-11-29 05:34:25+00:00  670838202509447168  670838202509447168   
2075 2015-11-29 05:16:59+00:00  670833812859932673  670833812859932672   

                                              full_text  truncated  \
0     This is Phineas. He's a mystical boy. Only eve...      False   
1     This is Tilly. She's just checking pup on you....      False   
2     This is Archie. He is a rare Norwegian Pouncin...      False   
3     This is Darla. She commenced a snooze mid meal...      False   
4     This is Franklin. He would like you to stop ca...      False   
...                                                 ...        ...   
2071  Yea I can't handle this job anymore your dogs ...      False   
2072  After so many requests... here you go.\n\nGood...      False   
2073  Meet Colby. He's that one cool friend that get...      False   
2074  Pink dogs here. Unreasonably long necks. Left ...      False   
2075  This is Jett. He is unimpressed by flower. 7/1...      False   

     display_text_range                                           entities  \
0               [0, 85]  {'hashtags': [], 'symbols': [], 'user_mentions...   
1              [0, 138]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2              [0, 121]  {'hashtags': [], 'symbols': [], 'user_mentions...   
3               [0, 79]  {'hashtags': [], 'symbols': [], 'user_mentions...   
4              [0, 138]  {'hashtags': [{'text': 'BarkWeek', 'indices': ...   
...                 ...                                                ...   
2071            [0, 93]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2072            [0, 81]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2073           [0, 139]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2074           [0, 140]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2075            [0, 71]  {'hashtags': [], 'symbols': [], 'user_mentions...   

                                      extended_entities  \
0     {'media': [{'id': 892420639486877696, 'id_str'...   
1     {'media': [{'id': 892177413194625024, 'id_str'...   
2     {'media': [{'id': 891815175371796480, 'id_str'...   
3     {'media': [{'id': 891689552724799489, 'id_str'...   
4     {'media': [{'id': 891327551943041024, 'id_str'...   
...                                                 ...   
2071  {'media': [{'id': 670995956771483648, 'id_str'...   
2072  {'media': [{'id': 670842759008411648, 'id_str'...   
2073  {'media': [{'id': 670840540850749440, 'id_str'...   
2074  {'media': [{'id': 670838194066300928, 'id_str'...   
2075  {'media': [{'id': 670833807587680256, 'id_str'...   

                                                 source  \
0     <a href="http://twitter.com/download/iphone" r...   
1     <a href="http://twitter.com/download/iphone" r...   
2     <a href="http://twitter.com/download/iphone" r...   
3     <a href="http://twitter.com/download/iphone" r...   
4     <a href="http://twitter.com/download/iphone" r...   
...                                                 ...   
2071  <a href="http://twitter.com/download/iphone" r...   
2072  <a href="http://twitter.com/download/iphone" r...   
2073  <a href="http://twitter.com/download/iphone" r...   
2074  <a href="http://twitter.com/download/iphone" r..

In [80]:
# 5- Remove letters from  name column in Twitter Archive Table
Invalid_Names = DF_TA_clean.name.str.contains('^[a-z]', regex = True)
DF_TA_clean.loc[Invalid_Names,'name'] = 'None'


In [81]:
# 6- Remove retweet entries from Twitter_Archive Table
DF_TA_clean = DF_TA_clean[DF_TA_clean.retweeted_status_id.isna()]

In [82]:
DF_TA_clean

tweet_id  in_reply_to_status_id  in_reply_to_user_id  \
0     892420643555336193                    NaN                  NaN   
1     892177421306343426                    NaN                  NaN   
2     891815181378084864                    NaN                  NaN   
3     891689557279858688                    NaN                  NaN   
4     891327558926688256                    NaN                  NaN   
...                  ...                    ...                  ...   
2071  671115716440031232                    NaN                  NaN   
2072  671109016219725825                    NaN                  NaN   
2073  670995969505435648                    NaN                  NaN   
2074  670842764863651840                    NaN                  NaN   
2075  670840546554966016                    NaN                  NaN   

       timestamp                                             source  \
0     2017-08-01  <a href="http://twitter.com/download/iphone" r...   
1     2017-08-01  <a href="http://twitter.com/download/iphone" r...   
2     2017-07-31  <a href="http://twitter.com/download/iphone" r...   
3     2017-07-30  <a href="http://twitter.com/download/iphone" r...   
4     2017-07-29  <a href="http://twitter.com/download/iphone" r...   
...          ...                                                ...   
2071  2015-11-29  <a href="http://twitter.com/download/iphone" r...   
2072  2015-11-29  <a href="http://twitter.com/download/iphone" r...   
2073  2015-11-29  <a href="http://twitter.com/download/iphone" r...   
2074  2015-11-29  <a href="http://twitter.com/download/iphone" r...   
2075  2015-11-29  <a href="http://twitter.com/download/iphone" r...   

                                                   text  retweeted_status_id  \
0     This is Phineas. He's a mystical boy. Only eve...                  NaN   
1     This is Tilly. She's just checking pup on you....                  NaN   
2     This is Archie. He is a rare Norwegian Pouncin...                  NaN   
3     This is Darla. She commenced a snooze mid meal...                  NaN   
4     This is Franklin. He would like you to stop ca...                  NaN   
...                                                 ...                  ...   
2071  Meet Phred. He isn't steering, looking at the ...                  NaN   
2072  This is Toby. He asked for chocolate cake for ...                  NaN   
2073  Yea I can't handle this job anymore your dogs ...                  NaN   
2074  After so many requests... here you go.\n\nGood...                  NaN   
2075  Meet Colby. He's that one cool friend that get...                  NaN   

      retweeted_status_user_id retweeted_status_timestamp  \
0                          NaN                        NaN   
1                          NaN                        NaN   
2                          NaN                        NaN   
3                          NaN                        NaN   
4                          NaN                        NaN   
...                        ...                        ...   
2071                       NaN                        NaN   
2072                       NaN                        NaN   
2073                       NaN                        NaN   
2074                       NaN                        NaN   
2075                       NaN                        NaN   

                                          expanded_urls  rating_numerator  \
0     https://twitter.com/dog_rates/status/892420643...                13   
1     https://twitter.com/dog_rates/status/892177421...                13   
2     https://twitter.com/dog_rates/status/891815181...                12   
3     https://twitter.com/dog_rates/status/891689557...                13   
4     https://twitter.com/dog_rates/status/891327558...                12   
...                                                 ...               ...   
2071  https://twitter.com/dog_rates/status/671115716...          

In [83]:
# 7- Remove reply entries from Twitter_Archive
DF_TA_clean = DF_TA_clean[DF_TA_clean.in_reply_to_status_id.isna()]

In [84]:
#Test
DF_TA_clean.groupby('retweeted_status_id').sum()

Empty DataFrame
Columns: [tweet_id, in_reply_to_status_id, in_reply_to_user_id, retweeted_status_user_id, rating_numerator, rating_denominator]
Index: []

In [85]:
#Test
DF_TA_clean.groupby('in_reply_to_status_id').sum()

Empty DataFrame
Columns: [tweet_id, in_reply_to_user_id, retweeted_status_id, retweeted_status_user_id, rating_numerator, rating_denominator]
Index: []

In [86]:
# 8- Rename the p1,p2,p3 columns labels
DF_IP_clean = DF_IP_clean.rename(columns={'p1':'Algorithm_Fisrt_Prediction','p2':'Algorithm_Second_Prediction','p3':'Algorithm_Third_Prediction'})

In [87]:
#Test
DF_IP_clean

tweet_id                                          jpg_url  \
0     666020888022790149  https://pbs.twimg.com/media/CT4udn0WwAA0aMy.jpg   
1     666029285002620928  https://pbs.twimg.com/media/CT42GRgUYAA5iDo.jpg   
2     666033412701032449  https://pbs.twimg.com/media/CT4521TWwAEvMyu.jpg   
3     666044226329800704  https://pbs.twimg.com/media/CT5Dr8HUEAA-lEu.jpg   
4     666049248165822465  https://pbs.twimg.com/media/CT5IQmsXIAAKY4A.jpg   
...                  ...                                              ...   
2070  891327558926688256  https://pbs.twimg.com/media/DF6hr6BUMAAzZgT.jpg   
2071  891689557279858688  https://pbs.twimg.com/media/DF_q7IAWsAEuuN8.jpg   
2072  891815181378084864  https://pbs.twimg.com/media/DGBdLU1WsAANxJ9.jpg   
2073  892177421306343426  https://pbs.twimg.com/media/DGGmoV4XsAAUL6n.jpg   
2074  892420643555336193  https://pbs.twimg.com/media/DGKD1-bXoAAIAUK.jpg   

      img_num Algorithm_Fisrt_Prediction   p1_conf  p1_dog  \
0           1     Welsh_springer_spaniel  0.465074    True   
1           1                    redbone  0.506826    True   
2           1            German_shepherd  0.596461    True   
3           1        Rhodesian_ridgeback  0.408143    True   
4           1         miniature_pinscher  0.560311    True   
...       ...                        ...       ...     ...   
2070        2                     basset  0.555712    True   
2071        1                paper_towel  0.170278   False   
2072        1                  Chihuahua  0.716012    True   
2073        1                  Chihuahua  0.323581    True   
2074        1                     orange  0.097049   False   

     Algorithm_Second_Prediction   p2_conf  p2_dog  \
0                         collie  0.156665    True   
1             miniature_pinscher  0.074192    True   
2                       malinois  0.138584    True   
3                        redbone  0.360687    True   
4                     Rottweiler  0.243682    True   
...                          ...       ...     ...   
2070            English_springer  0.225770    True   
2071          Labrador_retriever  0.168086    True   
2072                    malamute  0.078253    True   
2073                    Pekinese  0.090647    True   
2074                       bagel  0.085851   False   

       Algorithm_Third_Prediction   p3_conf  p3_dog  
0               Shetland_sheepdog  0.061428    True  
1             Rhodesian_ridgeback  0.072010    True  
2                      bloodhound  0.116197    True  
3              miniature_pinscher  0.222752    True  
4                        Doberman  0.154629    True  
...                           ...       ...     ...  
2070  German_short-haired_pointer  0.175219    True  
2071                      spatula  0.040836   False  
2072                       kelpie  0.031379    True  
2073                     papillon  0.068957    True  
2074                       banana  0.076110   False  

[2075 rows x 12 columns]

In [88]:
# 9- Rename column id name in tweet_data table
DF_TD_clean = DF_TD_clean.rename(columns={'id':'tweet_id'})

In [89]:
#Test
DF_TD_clean

created_at            tweet_id              id_str  \
0    2017-08-01 16:23:56+00:00  892420643555336193  892420643555336192   
1    2017-08-01 00:17:27+00:00  892177421306343426  892177421306343424   
2    2017-07-31 00:18:03+00:00  891815181378084864  891815181378084864   
3    2017-07-30 15:58:51+00:00  891689557279858688  891689557279858688   
4    2017-07-29 16:00:24+00:00  891327558926688256  891327558926688256   
...                        ...                 ...                 ...   
2071 2015-11-29 16:01:20+00:00  670995969505435648  670995969505435648   
2072 2015-11-29 05:52:33+00:00  670842764863651840  670842764863651840   
2073 2015-11-29 05:43:44+00:00  670840546554966016  670840546554966016   
2074 2015-11-29 05:34:25+00:00  670838202509447168  670838202509447168   
2075 2015-11-29 05:16:59+00:00  670833812859932673  670833812859932672   

                                              full_text  truncated  \
0     This is Phineas. He's a mystical boy. Only eve...      False   
1     This is Tilly. She's just checking pup on you....      False   
2     This is Archie. He is a rare Norwegian Pouncin...      False   
3     This is Darla. She commenced a snooze mid meal...      False   
4     This is Franklin. He would like you to stop ca...      False   
...                                                 ...        ...   
2071  Yea I can't handle this job anymore your dogs ...      False   
2072  After so many requests... here you go.\n\nGood...      False   
2073  Meet Colby. He's that one cool friend that get...      False   
2074  Pink dogs here. Unreasonably long necks. Left ...      False   
2075  This is Jett. He is unimpressed by flower. 7/1...      False   

     display_text_range                                           entities  \
0               [0, 85]  {'hashtags': [], 'symbols': [], 'user_mentions...   
1              [0, 138]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2              [0, 121]  {'hashtags': [], 'symbols': [], 'user_mentions...   
3               [0, 79]  {'hashtags': [], 'symbols': [], 'user_mentions...   
4              [0, 138]  {'hashtags': [{'text': 'BarkWeek', 'indices': ...   
...                 ...                                                ...   
2071            [0, 93]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2072            [0, 81]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2073           [0, 139]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2074           [0, 140]  {'hashtags': [], 'symbols': [], 'user_mentions...   
2075            [0, 71]  {'hashtags': [], 'symbols': [], 'user_mentions...   

                                      extended_entities  \
0     {'media': [{'id': 892420639486877696, 'id_str'...   
1     {'media': [{'id': 892177413194625024, 'id_str'...   
2     {'media': [{'id': 891815175371796480, 'id_str'...   
3     {'media': [{'id': 891689552724799489, 'id_str'...   
4     {'media': [{'id': 891327551943041024, 'id_str'...   
...                                                 ...   
2071  {'media': [{'id': 670995956771483648, 'id_str'...   
2072  {'media': [{'id': 670842759008411648, 'id_str'...   
2073  {'media': [{'id': 670840540850749440, 'id_str'...   
2074  {'media': [{'id': 670838194066300928, 'id_str'...   
2075  {'media': [{'id': 670833807587680256, 'id_str'...   

                                                 source  \
0     <a href="http://twitter.com/download/iphone" r...   
1     <a href="http://twitter.com/download/iphone" r...   
2     <a href="http://twitter.com/download/iphone" r...   
3     <a href="http://twitter.com/download/iphone" r...   
4     <a href="http://twitter.com/download/iphone" r...   
...                                                 ...   
2071  <a href="http://twitter.com/download/iphone" r...   
2072  <a href="http://twitter.com/download/iphone" r...   
2073  <a href="http://twitter.com/download/iphone" r...   
2074  <a href="http://twitter.com/download/iphone" r..

In [90]:
# 1- Removing unnecessary columns from Twitter Archive table
DF_TA_clean = DF_TA_clean.drop(['in_reply_to_status_id', 'in_reply_to_user_id', 'retweeted_status_id', 'retweeted_status_user_id', 'retweeted_status_timestamp'], axis=1)

In [91]:
# 2- Removing  unnecessary columns form tweet_data 
DF_TD_clean = DF_TD_clean.loc[:,['tweet_id','retweet_count','favorite_count']]
DF_TD_clean

tweet_id  retweet_count  favorite_count
0     892420643555336193           8853           39467
1     892177421306343426           6514           33819
2     891815181378084864           4328           25461
3     891689557279858688           8964           42908
4     891327558926688256           9774           41048
...                  ...            ...             ...
2071  670995969505435648            317            1175
2072  670842764863651840           4324            7989
2073  670840546554966016            213             634
2074  670838202509447168            758            1189
2075  670833812859932673            135             474

[2076 rows x 3 columns]

In [92]:
twitter_archive_master = pd.merge(DF_TA_clean, DF_TD_clean, how='inner' , on='tweet_id')

In [93]:
twitter_archive_master.head(5)

tweet_id   timestamp  \
0  892420643555336193  2017-08-01   
1  892177421306343426  2017-08-01   
2  891815181378084864  2017-07-31   
3  891689557279858688  2017-07-30   
4  891327558926688256  2017-07-29   

                                              source  \
0  <a href="http://twitter.com/download/iphone" r...   
1  <a href="http://twitter.com/download/iphone" r...   
2  <a href="http://twitter.com/download/iphone" r...   
3  <a href="http://twitter.com/download/iphone" r...   
4  <a href="http://twitter.com/download/iphone" r...   

                                                text  \
0  This is Phineas. He's a mystical boy. Only eve...   
1  This is Tilly. She's just checking pup on you....   
2  This is Archie. He is a rare Norwegian Pouncin...   
3  This is Darla. She commenced a snooze mid meal...   
4  This is Franklin. He would like you to stop ca...   

                                       expanded_urls  rating_numerator  \
0  https://twitter.com/dog_rates/status/892420643...                13   
1  https://twitter.com/dog_rates/status/892177421...                13   
2  https://twitter.com/dog_rates/status/891815181...                12   
3  https://twitter.com/dog_rates/status/891689557...                13   
4  https://twitter.com/dog_rates/status/891327558...                12   

   rating_denominator      name doggo floofer pupper puppo  retweet_count  \
0                  10   Phineas  None    None   None  None           8853   
1                  10     Tilly  None    None   None  None           6514   
2                  10    Archie  None    None   None  None           4328   
3                  10     Darla  None    None   None  None           8964   
4                  10  Franklin  None    None   None  None           9774   

   favorite_count  
0           39467  
1           33819  
2           25461  
3           42908  
4           41048

In [94]:
twitter_archive_master.to_csv('Twitter_Archive_Master.csv', index = False)